instrumentation/net/http/otelhttp/transport.go

```diff
@@ -97,6 +98,7 @@ func (t *Transport) RoundTrip(r *http.Request) (*http.Response, error) {
 	res, err := t.rt.RoundTrip(r)
 	if err != nil {
 		span.RecordError(err)
+		span.SetStatus(codes.Error, err.Error())
```

Comment:
Can we add tests to prevent regressions. This behavior used to exist in old versions of `RecordError`'s predecessor and I'm guessing it was intended by the original author. It would have been nice to have tests fail when it was removed upgrading to the latest `RecordError`. Adding tests now will help preserve this behavior in the future.

Reply:
I didn't add in the first place because apparently the coverage was good enough, and 100% coverage doesn't always mean better due to the complexity of use cases and the tests. Knowing that this is important enough, I added for both cases!